javascript/ql/src/semmle/javascript/security/dataflow/UnsafeShellCommandConstruction.qll

```diff
@@ -35,21 +36,11 @@ module UnsafeShellCommandConstruction {
     // override to require that there is a path without unmatched return steps
     override predicate hasFlowPath(DataFlow::SourcePathNode source, DataFlow::SinkPathNode sink) {
       super.hasFlowPath(source, sink) and
-      exists(DataFlow::MidPathNode mid |
-        source.getASuccessor*() = mid and
-        sink = mid.getASuccessor() and
-        mid.getPathSummary().hasReturn() = false
-      )
+      UnsafeHtmlConstruction::requireMatchedReturn(source, sink)
```

Comment:
Can we move this step to a shared library. 
I the step already existed in `UnsafeShellCommandConstruction.qll `, and was simply referenced from `UnsafeHtmlConstruction.qll` I would be more inclined to shrug this off, but now we are introducing a dependency on a new query instead.